src/App/Repository/HXL/HXLTagRepository.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * Ushahidi HXLTag Repository, using Kohana::$config
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Repository\HXL;
+
+use Ohanzee\Database;
+use Ushahidi\Core\SearchData;
+use Ushahidi\Core\Entity\HXL\HXLTag;
+use Ushahidi\Core\Entity\HXL\HXLTagRepository as HXLTagRepositoryContract;
+use Ushahidi\Core\Usecase\ReadRepository;
+use Ushahidi\Core\Usecase\SearchRepository;
+use Ushahidi\App\Repository\OhanzeeRepository;
+use Ohanzee\DB;
+
+class HXLTagRepository extends OhanzeeRepository implements
+	HXLTagRepositoryContract,
+	SearchRepository,
+	ReadRepository
+{
+	private $tags_attributes;
+
+	public function __construct(Database $db)
+	{
+		parent::__construct($db);
+	}
+
+	// OhanzeeRepository
+	protected function getTable()
+	{
+		return 'hxl_tags';
+	}
+
+	public function getSearchFields()
+	{
+		return ['tag_name'];
+	}
+
+	public function setSearchConditions(SearchData $search)
+	{
+		$query = $this->search_query;
+		return $query;
+	}
+
+	/**
+	 * @param array|null $data
+	 * @return \Ushahidi\App\Repository\Ushahidi\Core\Entity|HXLTag|\Ushahidi\Core\Usecase\Entity
+	 */
+	public function getEntity(array $data = null)
+	{
+		if (!$this->tags_attributes) {
+			$this->tags_attributes = $this->getAllHXLAttributes();
+		}
+
+		if ($data) {
+			$data['form_attribute_types'] = $this->getFormAttributeTypes($data['id']);
+			$data['hxl_attributes'] = array_filter($this->tags_attributes, function ($tag_attributes) use ($data) {
```

Comment:
Would replace this with $this->getAttributes($tag_id) and delegate all the filtering etc to there